tests/deepspeed/test_ulysses.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import sys
+
+import torch
+from deepspeed import initialize
```

Comment:
These both need to be guarded under their respective `if_xxx_available`, it's why the current tests are failing

Reply:
The test (in top-level test_deepspeed.py) is already guarded `@required_deepspeed` and [@required_torch_accelerator](https://github.com/huggingface/transformers/blob/9ea2571e79729616c86a6e5dc0ab942a8f4447fc/tests/deepspeed/test_deepspeed.py#L1339). I am thinking that should take of it, no?

Reply:
Nope as it's being imported, we need to guard it from regular import checks.